### DISCUSSION TITLE
Video controls

### DISCUSSION THREAD

**Discussion by agloolik** [react-page/react-page#1286] [Q&A]
Hi Everyone,
How do you pass a prop to the video renderer so that in react-player components controls={true}.
If I add a vimeo video by default it will not play or show controls.
Any help appreciated, many Thanks !

**Comment by macrozone** [comment]
i suggest to read the docs about custom-cell-plugins https://react-page.github.io/docs/#/custom-cell-plugins

then create a cell-plugin to your needs. usually a video just needs a video-id (e.g. from youtube or vimeo). The existing video plugin is a bit old, its better and maybe easier to just create a custom one (its really simple actually)

**Comment by agloolik** [comment]
Thanks macrozone, I did a copy of the video plugin that I slightly modified, I should definitely spend more time to go more in depth with this project !

**Comment by agloolik** [comment]
[video.zip](https://github.com/react-page/react-page/files/13377577/video.zip)
You have to import the file

import video from "../../plugins/video";

 and then include it in the editor

  const cellPlugins = useMemo(
    () => [
      slate(),
      spacer,
      imagePlugin({ imageUpload: imageUpload("/images/sea-bg.jpg") }),
      video,
      divider,
      html5video,
      background({
        imageUpload: imageUpload("/images/sea-bg.jpg"),
        enabledModes:
          ModeEnum.COLOR_MODE_FLAG |
          ModeEnum.IMAGE_MODE_FLAG |
          ModeEnum.GRADIENT_MODE_FLAG,
      }),
    ],
    []
  );
  return (
      <Editor
        name="content"
        type="JSON"
        cellPlugins={cellPlugins}
        languages={LANGUAGES}
        value={value}
        lang={LANGUAGES[0].lang}
        onChange={onChange}
        readOnly={readOnly}
      />
)

**Reply by agloolik** [reply]
`import video from "../../plugins/video";`

`  const cellPlugins = useMemo(
    () => [
      slate(),
      spacer,
      imagePlugin({ imageUpload: imageUpload("/images/sea-bg.jpg") }),
      video,
      divider,
      html5video,
      background({
        imageUpload: imageUpload("/images/sea-bg.jpg"),
        enabledModes:
          ModeEnum.COLOR_MODE_FLAG |
          ModeEnum.IMAGE_MODE_FLAG |
          ModeEnum.GRADIENT_MODE_FLAG,
      }),
    ],
    []
  );
  return (
      <Editor
        name="content"
        type="JSON"
        cellPlugins={cellPlugins}
        languages={LANGUAGES}
        value={value}
        lang={LANGUAGES[0].lang}
        onChange={onChange}
        readOnly={readOnly}
      />
  );
`

**Reply by agloolik** [reply]
It seems I can't format this right sorry about that :)

**Reply by katerynapadafa** [reply]
i have solved this issue like that:
```import React from 'react';
import PropTypes from 'prop-types';
import video from '@react-page/plugins-video';
import ReactPlayer from 'react-player';

const customVideoPlugin = {
  ...video,
  Renderer: props => (
    <div
      style={{
        alignContent: 'center',
        alignItems: 'center',
        position: 'relative',
        height: '0px',
        paddingBottom: '65.25%',
      }}
    >
      {props.readOnly ? null : (
        <div
          style={{
            position: 'absolute',
            top: 0,
            left: 0,
            right: 0,
            bottom: 0,
            zIndex: 10,
          }}
        />
      )}
      <ReactPlayer
        url={props.data ? props.data.src : undefined}
        height="100%"
        width="100%"
        style={{
          position: 'absolute',
          width: '100%',
          height: '100%',
        }}
        blur={props.data ? props.data.blur : undefined}
        controls
        muted={props.data ? props.data.muted : undefined}
        playing={false}
        loop={props.data ? props.data.loop : undefined}
      />
    </div>
  ),
};

customVideoPlugin.Renderer.propTypes = {
  data: PropTypes.shape({
    src: PropTypes.string,
    blur: PropTypes.bool,
    muted: PropTypes.bool,
    loop: PropTypes.bool,
  }),
  readOnly: PropTypes.bool,
};

export default customVideoPlugin;
```
